### ISSUE TITLE
Mapping update for “date_range” field type is not idempotent

### ISSUE THREAD

**Issue by roman-qu** [opensearch-project/OpenSearch#1921]
**Describe the bug**
OS does not allow to update date_range fields with exact same mapping settings and responds with 500.
That is known issue for ElasticSearch version 7.10 which OpenSearch used for fork. It was resolved in version 7.11. Link to issue created for ElasticSearch -> https://github.com/elastic/elasticsearch/issues/69012

**To Reproduce**
1. Create an index `curl -XPUT http://localhost:9200/test_index`
2. Add a mapping for the field with date_range type. 
`curl -XPUT http://localhost:9200/test_index/_mapping \
-H 'Content-Type: application/json' \
-d '{"properties": {"range": {"type": "date_range", "format": "yyyy-MM-dd"}}}'`

3. Update the same field using the same mapping settings - repeat the previous request 
```
curl -XPUT http://localhost:9200/test_index/_mapping \
-H 'Content-Type: application/json' \
-d '{"properties": {"range": {"type": "date_range", "format": "yyyy-MM-dd"}}}'

#response:
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_state_exception",
        "reason" : "[index.version.created] is not present in the index settings for index with UUID [null]"
      }
    ],
    "type" : "illegal_state_exception",
    "reason" : "[index.version.created] is not present in the index settings for index with UUID [null]"
  },
  "status" : 500
}
```

**Expected behavior**
Should allow to update existed mapping.

**Plugins**
No extra plugins

**Host/Environment (please complete the following information):**
 - OpenSearch version 1.2.3 and 1.1.0

**Additional context**
We encountered that issue during using the dynamic_templates feature in combination with date_range fields. Under the hood, the Dynamic templates feature performs the same field mapping update, but implicitly, while indexing a new document.
Issue might be fixed in 7.11 version as part of this refactoring https://github.com/elastic/elasticsearch/pull/64635

**Comment by dblock** [comment]
Which versions did we release this fix in? Can't seem to find it in release notes.

**Comment by andrross** [comment]
@dblock From browsing the commit history, it was wirst released in [1.3.0](https://github.com/opensearch-project/OpenSearch/commits/1.3.0?after=e45991597c86ba1bbcc36ee1dfdc165197a913af+69&branch=1.3.0&qualified_name=refs%2Ftags%2F1.3.0) an subsequently made it into 2.0 as well.

**Comment by andrross** [comment]
It is in the [1.3.0 release notes](https://github.com/opensearch-project/OpenSearch/blob/main/release-notes/opensearch.release-notes-1.3.0.md?plain=1#L350) as well.
